Assets/Plugins/iOS/Shopify/WebCheckoutViewController.swift

```diff
@@ -0,0 +1,345 @@
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func loadView() {
+        self.view = WebCheckoutView(frame: UIScreen.main.bounds)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        webView.navigationDelegate = self
+        checkoutView.actionButton.addTarget(self, action: #selector(didPressCancel), for: .touchUpInside)
+        
+        loadCheckoutURL()
+    }
+}
+
+// MARK: Actions
+extension WebCheckoutViewController {
+    func loadCheckoutURL() {
+        let request = URLRequest(url: url)
+        webView.load(request)
+    }
+    
+    func didPressCancel() {
+        delegate?.willDismiss()
+        self.dismiss(animated: true)
+    }
+}
+
+// MARK: WKNavigationDelegate
+extension WebCheckoutViewController: WKNavigationDelegate {
+    public func webView(_ webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!) {
+        checkoutView.showProgressIndicator()
+    }
+    
+    public func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {}
+    
+    public func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
+        if let url = webView.url, isThankYouPage(url) {
+            delegate?.didLoadThankYouPage()
+            self.checkoutView.isFinished = true
+            return
+        }
+    }
+    
+    public func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        checkoutView.hideProgressIndicator()
+    }
+    
+    public func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: Error) {}
+    
+    public func webView(_ webView: WKWebView, didReceive challenge: URLAuthenticationChallenge,
+                        completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Swift.Void) {
+        completionHandler(.performDefaultHandling, nil)
+    }
+    
+    public func webViewWebContentProcessDidTerminate(_ webView: WKWebView) {}
+    
+    private func isThankYouPage(_ url: URL) -> Bool {
+        return url.absoluteString.hasSuffix("/thank_you")
+    }
+}
+
+// MARK: Views
+
+// View contents of the checkout view controller.
+private class WebCheckoutView: UIView {
+    var isFinished = false {
+        didSet {
+            actionButton.checkoutState = isFinished ? .finished : .checkingOut
+        }
+    }
+    
+    private let cornerRadius: CGFloat = 3
+    private let viewMargin: CGFloat = 10
+    private let buttonHeight: CGFloat = 46
+    private let loadingIndicatorOffset: CGFloat = 5
+    private let loadingIndicatorTiming: TimeInterval = 0.3
+    
+    private(set) var containerView: UIView!
+    private(set) var webView: WKWebView!
+    private(set) var actionButton: CheckoutButton!
+    private(set) var progressIndicator: WebPageProgressIndicator!
```

Comment:
Nope! Was used to declaring them on a view controller but since we set them up in `init` it's good.